### PR TITLE
Fix wrong behaviors when handling sub req

### PR DESCRIPTION
**Pull request by wanghaEMQ**

### CODE REVIEW

File: nanomq/sub_handler.c [hunk 1]
```diff
@@ -146,12 +148,22 @@ encode_suback_msg(nng_msg *msg, nano_work *work)
...
+	reason_code = PACKET_IDENTIFIER_IN_USE;
+	if (sub_pkt->packet_id == 0) {
+		if ((rv = nng_msg_append(msg, &reason_code, 1)) != 0) {
+			debug_msg("ERROR: nng_msg_append [%d]", rv);
+			return PROTOCOL_ERROR;
+		}
+	}
+
+	// Note. When packet_id is zero, node must be empty. So, Dont worry
+	// the order of reasone codes would be changed.
```
Comment (lee-emqx) [inline nanomq/sub_handler.c:166]: reason

**Review by lee-emqx — APPROVED** [review]

**Comment by JaylinYu** [comment]
Plz check CodeQL warnings

File: nanomq/sub_handler.c (hunk 1 above)
Comment (wanghaEMQ) [reply nanomq/sub_handler.c:166]: That's how I design it. Nodes would not be allocated if packet id is equal to zero. So we just respond an ack to remind the client that the packet id is wrong.

**Comment by wanghaEMQ** [comment]
> will merge after all the critical warnings are gone

All the critical warnings are comes from the files I don't modify

**Review by JaylinYu — APPROVED** [review]

**Pull request merged**